File: Lab2/input1.py
def analyzeNumberProperties(n):
    print("Analyzing number: ", n)
    if n <= 0:
        print("Only positive integers are analyzed.")
        return
    sum = 0
    divisorCount = 0
    evenDivisors = 0
    oddDivisors = 0
    primeDivisors = 0
    print("Finding all proper divisors of ", n, "...")
    for i in range(1, n):
        if n % i == 0:
            sum += i
            divisorCount += 1
            print("Found divisor: ", i)
            if i % 2 == 0:
                evenDivisors += 1
            else:
                oddDivisors += 1
            isPrime = True
            if i < 2:
                isPrime = False
            else:
                for j in range(2, int(i ** 0.5) + 1):
                    if i % j == 0:
                        isPrime = False
                        break
            if isPrime:
                primeDivisors += 1
                print("  -> ", i, " is a prime divisor.")
    print("=== SUMMARY ===")
    print("Sum of proper divisors: ", sum)
    print("Total divisors: ", divisorCount)
    print("Even divisors: ", evenDivisors)
    print("Odd divisors: ", oddDivisors)
    print("Prime divisors: ", primeDivisors)
    if sum == n:
        print(n, " is a PERFECT number!")
    elif sum > n:
        print(n, " is an ABUNDANT number.")
    else:
        print(n, " is a DEFICIENT number.")
    if divisorCount == 1 and n > 1:
        print(n, " is also a PRIME number.")

File: Lab2/Translator_MPL/input1.cpp
#include <iostream>
using namespace std;

void analyzeNumberProperties(int n) {
    cout << "Analyzing number: " << n << endl;
    if (n <= 0) {
        cout << "Only positive integers are analyzed." << endl;
        return;
    }
    int sum = 0;
    int divisorCount = 0;
    int evenDivisors = 0;
    int oddDivisors = 0;
    int primeDivisors = 0;
    cout << "Finding all proper divisors of " << n << "..." << endl;
    for (int i = 1; i < n; i++) {
        if (n % i == 0) {
            sum += i;
            divisorCount++;
            cout << "Found divisor: " << i << endl;
            if (i % 2 == 0) {
                evenDivisors++;
            } else {
                oddDivisors++;
            }
            bool isPrime = true;
            if (i < 2) {
                isPrime = false;
            } else {
                for (int j = 2; j * j <= i; j++) {
                    if (i % j == 0) {
                        isPrime = false;
                        break;
                    }
                }
            }
            if (isPrime) {
                primeDivisors++;
                cout << "  -> " << i << " is a prime divisor." << endl;
            }
        }
    }
    cout << "=== SUMMARY ===" << endl;
    cout << "Sum of proper divisors: " << sum << endl;
    cout << "Total divisors: " << divisorCount << endl;
    cout << "Even divisors: " << evenDivisors << endl;
    cout << "Odd divisors: " << oddDivisors << endl;
    cout << "Prime divisors: " << primeDivisors << endl;
    if (sum == n) {
        cout << n << " is a PERFECT number!" << endl;
    } else if (sum > n) {
        cout << n << " is an ABUNDANT number." << endl;
    } else {
        cout << n << " is a DEFICIENT number." << endl;
    }
    if (divisorCount == 1 && n > 1) {
        cout << n << " is also a PRIME number." << endl;
    }
    cout << "Analysis complete for " << n << "." << endl;
}